File: src/main/java/com/smarthome/backend/model/devices/helper/DeviceType.java
package com.smarthome.backend.model.devices.helper;

/**
 * Enum für die verschiedenen Gerätetypen im Smart Home System.
 */
public enum DeviceType {
    LIGHT("light"),
    LIGHT_DIMMER("light-dimmer"),
    LIGHT_DIMMER_TEMPERATURE_COLOR("light-dimmer-temperature-color"),
    LIGHT_DIMMER_TEMPERATURE("light-dimmer-temperature"),
    SWITCH("switch"),
    SWITCH_DIMMER("switch-dimmer"),
    THERMOSTAT("thermostat"),
    MOTION("motion"),
    LIGHT_LEVEL("light-level"),
    TEMPERATURE("temperature"),
    MOTION_LIGHT_LEVEL("motion-light-level"),
    MOTION_LIGHT_LEVEL_TEMPERATURE("motion-light-level-temperature"),
    SENSOR("sensor"),
    VACUUM("vacuum"),
    SPEAKER("speaker"),
    CAR("car"),
    LOCK("lock"),
    TV("tv");
    
    private final String value;
    
    DeviceType(String value) {
        this.value = value;
    }
    
    /**
     * Gibt den String-Wert des Enums zurück (für JSON-Serialisierung).
     * 
     * @return Der String-Wert des Gerätetyps
     */
    public String getValue() {
        return value;
    }
    
    /**
     * Konvertiert einen String-Wert in das entsprechende DeviceType-Enum.
     * 
     * @param value Der String-Wert (z.B. "light", "switch", "speaker", etc.)
     * @return Das entsprechende DeviceType-Enum oder null, wenn der Wert nicht erkannt wird
     */
    public static DeviceType fromString(String value) {
        if (value == null) {
            return null;
        }
        for (DeviceType type : DeviceType.values()) {
            if (type.value.equals(value)) {
                return type;
            }
        }
        return null;
    }
}

File: src/main/node/com/smarthome/backend/model/devices/helper/DeviceType.ts
/**
 * DeviceType Enum - Alle unterstützten Gerätetypen.
 * Kompatibel mit Frontend-Datenmodell.
 */
export enum DeviceType {
  // Licht-Geräte
  LIGHT = "light",
  LIGHT_DIMMER = "light-dimmer",
  LIGHT_DIMMER_TEMPERATURE = "light-dimmer-temperature",
  LIGHT_DIMMER_TEMPERATURE_COLOR = "light-dimmer-temperature-color",
  
  // Schalter-Geräte
  SWITCH = "switch",
  SWITCH_DIMMER = "switch-dimmer",
  SWITCH_ENERGY = "switch-energy",
  OUTLET = "outlet",
  
  // Sensoren
  MOTION = "motion",
  LIGHT_LEVEL = "light-level",
  TEMPERATURE = "temperature",
  MOTION_LIGHT_LEVEL = "motion-light-level",
  MOTION_LIGHT_LEVEL_TEMPERATURE = "motion-light-level-temperature",
  SENSOR = "sensor",
  
  // Klima & Komfort
  THERMOSTAT = "thermostat",
  BLINDS = "blinds",
  
  // Sicherheit
  LOCK = "lock",
  CAMERA = "camera",
  DOORBELL = "doorbell",
  
  // Unterhaltung
  SPEAKER = "speaker",
  SPEAKER_RECEIVER = "speaker-receiver",
  TV = "tv",
  
  // Haushaltsgeräte
  VACUUM = "vacuum",
  IRRIGATION = "irrigation",
  GARAGE = "garage",
  
  // Fahrzeuge
  CAR = "car"
}

export function deviceTypeFromString(value?: string | null) {
  if (!value) return null;
  const entries = Object.values(DeviceType) as string[];
  return entries.includes(value) ? (value as DeviceType) : null;
}

